CONTRIBUTING.md

```diff
@@ -320,6 +324,38 @@ make scaffold name=streamInstance type=resource
 
 This will generate resource/data source files and accompanying test files needed for starting the development, and will contain multiple comments with `TODO:` statements which give guidance for the development.
 
+#### Scaffolding Schema and Model Definitions
+
+Complementary to the `scaffold` command, there is a command which generates the initial Terraform schema definition and associated Go types for a resource or data source. This processes leverages [Code Generation Tools](https://developer.hashicorp.com/terraform/plugin/code-generation) developed by HashiCorp, which in turn make use of the [Atlas Admin API](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/) OpenAPI Specification.
+
+##### Running the command
+
+Both `tfplugingen-openapi` and `tfplugingen-framework` need to be installed which can be done by running `make tools`.
+
+The command takes a single argument which specifies for the resource or data source where the code generation is run, defined in camel case:
+```bash
+make scaffold-schemas name=streamInstance
+```
+
+As a pre-requiste, the relevant resource/data source directory must define a configuration file in the path `./internal/service/<resource_name>/tfplugingen/generator_config.yml`. The content of this file will define which resource and/or data sources schemas will be generated by providing the API endpoints they are mapped to. You can reference the [Generator Config](https://developer.hashicorp.com/terraform/plugin/code-generation/openapi-generator#generator-config) documentation to gain more insight on configuration options. An example defined in our repository can be found in [searchdeployment/tfplugingen/generator_config.yml](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/internal/service/searchdeployment/tfplugingen/generator_config.yml).
+
+As a result of the execution, the schema definitions and associated model types will be defined in separate files depending on resource and data sources that were configured in the generator_config.yml file:
+- `data_source_<resource_name>_schema.go`
+- `resource_<resource_name>_schema.go`
+
+Note: if the resulting file paths already exist the content will be stored in files with a `_gen.go` postfix, and in this case any content will be overwritten. This can be useful for comparing the lastest autogenerated schema againt the existing implementation.
```

Comment:
```suggestion
Note: if the resulting file paths already exists the content will be stored in files with a `_gen.go` postfix, and in this case any content will be overwritten. This can be useful for comparing the lastest autogenerated schema againt the existing implementation.
```